src/SignalR/server/Core/src/Internal/DefaultHubDispatcher.cs

```diff
@@ -27,14 +26,43 @@ internal partial class DefaultHubDispatcher<THub> : HubDispatcher<THub> where TH
         private readonly IHubContext<THub> _hubContext;
         private readonly ILogger<HubDispatcher<THub>> _logger;
         private readonly bool _enableDetailedErrors;
+        private readonly Func<HubInvocationContext, ValueTask<HubResult>> _invokeMiddleware;
+        private readonly Func<HubInvocationContext, Task> _onConnectedMiddleware;
+        private readonly Func<HubInvocationContext, Exception, Task> _onDisconnectedMiddleware;
 
-        public DefaultHubDispatcher(IServiceScopeFactory serviceScopeFactory, IHubContext<THub> hubContext, bool enableDetailedErrors, ILogger<DefaultHubDispatcher<THub>> logger)
+        public DefaultHubDispatcher(IServiceScopeFactory serviceScopeFactory, IHubContext<THub> hubContext, bool enableDetailedErrors,
+            ILogger<DefaultHubDispatcher<THub>> logger, List<IHubFilter> hubFilters)
         {
             _serviceScopeFactory = serviceScopeFactory;
             _hubContext = hubContext;
             _enableDetailedErrors = enableDetailedErrors;
             _logger = logger;
             DiscoverHubMethods();
+
+            var count = hubFilters?.Count ?? 0;
+            if (count != 0)
+            {
+                _invokeMiddleware = async (invocationContext) =>
+                {
+                    return HubResult.WithResult(await ExecuteMethod(invocationContext.HubMethod.Name, invocationContext.Hub, invocationContext.HubMethodArguments.ToArray()));
```

Comment:
ToArray()? We allocate a new array everytime? we need to fix this. We should be reusing the arguments if it hasn't been modified (copy on write).